pkg/infra/usagestats/statscollector/service_test.go

```diff
@@ -329,8 +330,8 @@ func TestDatasourceStats(t *testing.T) {
 		assert.EqualValues(t, 1, dba["stats.ds_access."+datasources.DS_ES+".direct.count"])
 		assert.EqualValues(t, 2, dba["stats.ds_access."+datasources.DS_ES+".proxy.count"])
 		assert.EqualValues(t, 3, dba["stats.ds_access."+datasources.DS_PROMETHEUS+".proxy.count"])
-		assert.EqualValues(t, 6+7, dba["stats.ds_access.other.direct.count"])
```

Comment:
I think it would be good to still demonstrate that if the validator returns `false`, that these stats would be populated. You might just update the Fake struct to make life easier:

```go
type FakeUsageStatsValidator struct{
   ShouldBeReportedFunc func(ctx context.Context, dsType string) bool
}

func (uss *FakeUsageStatsValidator) ShouldBeReported(ctx context.Context, dsType string) bool {
   if uss.ShouldBeReportedFunc != nil {
       return uss.ShouldBeReportedFunc(ctx, dsType) 
   }
   return true
}
```